test/provenance.cpp

```diff
@@ -335,6 +335,56 @@ TEST(provenance, add_group_above)
     EXPECT_EQ(m1->get_provenance_tags(), (ProvSet{"m1"}));
 }
 
+TEST(provenance, add_tags_above)
+{
+    auto x = make_shared<op::Parameter>(element::i32, PartialShape{2, 3, 4});
+    auto y = make_shared<op::Parameter>(element::i32, PartialShape{2, 3, 4});
+
+    auto a = make_shared<op::Add>(x, y);
+    auto b = make_shared<op::Multiply>(x, y);
+    auto c = make_shared<op::Subtract>(a, b);
+    auto d = make_shared<op::Abs>(c);
+
+    // Add tags to Subtract and all nodes until Parameters (all above c, until params x, y)
+    c->add_provenance_tags_above(OutputVector{x, y}, {"tag_above_c - until_params"});
+    // Add tags to Abs and Subtract (above d, until c inputs)
+    d->add_provenance_tags_above(c->input_values(), {"tag_above_d - until_c_inputs"});
+    // Add tags to Abs and all nodes above
+    d->add_provenance_tags_above(OutputVector{}, {"tag_all_above_d"});
+
+    auto x_tags = x->get_provenance_tags();
+    EXPECT_EQ(x_tags.size(), 1);
+    EXPECT_EQ(x_tags.find("tag_all_above_d") != x_tags.end(), true);
+
+    auto y_tags = y->get_provenance_tags();
+    EXPECT_EQ(y_tags.size(), 1);
+    EXPECT_EQ(y_tags.find("tag_all_above_d") != y_tags.end(), true);
+
+    auto a_tags = a->get_provenance_tags();
+    EXPECT_EQ(a_tags.size(), 2);
+    EXPECT_EQ(a_tags.find("tag_above_c - until_params") != a_tags.end(), true);
```

Comment:
In UT code it's customary to use `EXPECT_TRUE` and `EXPECT_FALSE` to express this kind of conditions. This is just a suggestion, not a request to change it.

Reply:
Nice hint, thanks